gateway-js/src/loadCsdlFromStorage.ts

```diff
@@ -0,0 +1,102 @@
+import { fetch } from 'apollo-server-env';
+import { GraphQLError } from 'graphql';
+import { CsdlQuery } from './__generated__/graphqlTypes';
+
+export const CSDL_QUERY = /* GraphQL */`#graphql
+  query Csdl($apiKey: String!, $ref: String!) {
+    routerConfig(ref: $ref, apiKey: $apiKey) {
+      __typename
+      ... on RouterConfigResult {
+        id
+        csdl
+      }
+      ... on FetchError {
+        code
+        message
+      }
+    }
+  }
+`;
+
+type CsdlQueryResult = CsdlQuerySuccess | CsdlQueryFailure;
+
+interface CsdlQuerySuccess {
+  data: CsdlQuery;
+}
+
+interface CsdlQueryFailure {
+  data?: CsdlQuery;
+  errors: GraphQLError[];
+}
+
+const { name, version } = require('../package.json');
+
+export async function loadCsdlFromStorage({
+  graphId,
+  graphVariant,
+  apiKey,
+  endpoint,
+  fetcher,
+}: {
+  graphId: string;
+  graphVariant: string;
+  apiKey: string;
+  endpoint: string;
+  fetcher: typeof fetch;
+}) {
+  const result = await fetcher(endpoint, {
```

Comment:
I suspect we might want more detail on errors thrown here (which could be network errors, issues with our API availability, etc). ie just wrapping in something saying that it was an error loading configuration from Apollo or whatever. Though that could be added later.

Reply:
9b2e84e